examples/ios/PickerView.swift

```diff
@@ -0,0 +1,267 @@
+//
+//  PickerView.swift
+//  React Native Segmented Picker
+//
+//  Created by Adam McArthur on 20/6/20.
+//
+
+import UIKit
+
+class PickerView: UIView, UIPickerViewDelegate, UIPickerViewDataSource {
+  let picker: UIPickerView = UIPickerView()
+  var options = [PickerColumn]()
+  var defaultSelectionsApplied: Bool = false
+  var theme: PickerTheme = PickerTheme(theme: nil)
+
+  // React Props
+  @objc(options) var optionsProp: [NSDictionary] = [] {
+    didSet {
+      parseOptionsProp()
+    }
+  }
+  @objc(defaultSelections) var defaultSelectionsProp = [String: String]() {
+    didSet {
+      applyDefaultSelections()
+    }
+  }
+  @objc(theme) var themeProp = NSDictionary() {
+    didSet {
+      theme = PickerTheme(theme: themeProp)
+    }
+  }
+  @objc var onValueChange: RCTBubblingEventBlock?
+  @objc var onEmitSelections: RCTBubblingEventBlock?
+
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    picker.dataSource = self
+    picker.delegate = self
+    self.addSubview(picker)
+  }
+
+  /**
+   * Setup the picker frame when the view appears on the screen.
+   */
+  override func willMove(toWindow newWindow: UIWindow?) -> Void {
+    super.willMove(toWindow: newWindow)
+    // AKA: View Did Appear
+    if newWindow != nil {
+      picker.frame = self.frame
+      let selectionHighlighter = UIView(frame: CGRect(
+        x: 0,
+        y: 0,
+        width: picker.frame.size.width,
+        height: theme.itemHeight
+      ))
+      selectionHighlighter.backgroundColor = UIColor.black.withAlphaComponent(
+        theme.selectionHighlightAlpha
+      )
+      selectionHighlighter.center = CGPoint(
+        x: picker.frame.size.width / 2,
+        y: picker.frame.size.height / 2
+      )
+      picker.addSubview(selectionHighlighter)
+    }
+  }
```

Comment:
Is there a better way of doing this? I was looking for something similar to `viewDidLoad`.

Reply:
Also - what this method is doing is essentially adding a highlight colour between the two selection bar indicators in the `UIPickerView`. One issue that I haven't been able to solve yet is how to get this view to add itself _behind_ the currently selected picker label. This code above actually adds it in front.

I tried messing around with a whole bunch of zIndex stuff, but to no avail. I take it "sub views" have their own funny little z index ordering hierarchy...

Reply:
Figured this one out. I needed to add the selection highlighter in a subview that was one deeper that what I was previously adding to.